nmostesting/suites/IS1301Test.py

```diff
@@ -0,0 +1,316 @@
+    For each annotable object type (label, description, tags):
+        - Read initial value
+            - store
+        - Reset default value by sending null
+            - check value + timestamp + is-14/value + is-04 timestamp
+            - store
+        - Write max-length
+            - check value + timestamp + is-14/value + is-04 timestamp
+        - Write >max-length
+            - check value + timestamp + is-14/value + is-04 timestamp
+        - Reset default value again
+            - check value + timestamp + is-14/value + is-04 timestamp
+            - compare with 1st reset
+        - Restore initial value
+"""
+
+from ..GenericTest import GenericTest, NMOSTestException
+from ..TestHelper import compare_json
+
+from ..import TestHelper
+import re
+import copy
+import time
+
+ANNOTATION_API_KEY = "annotation"
+NODE_API_KEY = "node"
+
+RESOURCES = ["self", "devices", "senders", "receivers"]
+OBJECTS = ["label", "description", "tags"]
+
+# const for label&description-related tests
+STRING_OVER_MAX_VALUE = ''.join(['X' for i in range(100)])
+STRING_MAX_VALUE = STRING_OVER_MAX_VALUE[:64]  # this is the max length tolerated
+
+# const for tags-related tests
+TAGS_OVER_MAX_VALUE = {'location': ['underground'], 'studio': ['42'], 'tech': ['John', 'Mike']}
+TAGS_MAX_VALUE = TAGS_OVER_MAX_VALUE.copy()
+TAGS_OVER_MAX_VALUE.pop('tech')  # must have a max of 5
+TAGS_TO_BE_SKIPPED = 'urn:x-nmos:tag:grouphint/v1.0'
+
+
+def get_ts_from_version(version):
+    """ Convert the 'version' object (string) into float """
+    return float(re.sub(':', '.', version))
+
+
+def strip_tags(tags):
+    if TAGS_TO_BE_SKIPPED in list(tags.keys()):
+        tags.pop(TAGS_TO_BE_SKIPPED)
+    return tags
+
+
+class IS1301Test(GenericTest):
+    """
+    Runs IS-13-Test
+    """
+
+    def __init__(self, apis, **kwargs):
+        GenericTest.__init__(self, apis, **kwargs)
+        self.annotation_url = self.apis[ANNOTATION_API_KEY]["url"]
+        self.node_url = f"{self.apis[ANNOTATION_API_KEY]['base_url']}/x-nmos/node/{self.apis[NODE_API_KEY]['version']}/"
+
+    def set_up_tests(self):
+        """
+        FAKE_ORIG = {
+            'description': 'fake_orig_desc',
+            'label': 'fake_orig_label',
+            'tags': {
+                'location': ['fake_location']
+            }
+        }
+        for resource in RESOURCES:
+            url = "{}{}{}".format(self.annotation_url, 'node/', resource)
+            TestHelper.do_request("PATCH", url, json=FAKE_ORIG)
+        """
+        pass
+
+    def get_resource(self, url):
+        """ Get a resource """
+
+        valid, r = self.do_request("GET", url)
+        if valid and r.status_code == 200:
+            try:
+                return True, r.json()
+            except Exception as e:
+                return False, e.msg
+        else:
+            return False, "GET  Resquest FAIL"
+
+    def compare_resource(self, object, new, resp):

```

Comment:
I guess `resp` here is response, which doesn't make mush sense in the context of a comparison function.

Would prefer this to be consistent with other compare functions in the code base (e.g. `NMOSUtils.compare_urls`, `NMOSUtils.compare_api_version` etc.) that, (rightly or wrongly) use `(url1, url2)` and `(ver1, ver2)`, so `(res1, res2)` might be preferable

Reply:
Also `object` both an overly broad and overloaded term; is there something more descriptive that could be used here? `tag_type`?